pkg/ccl/backupccl/key_rewriter.go

```diff
@@ -146,10 +164,29 @@ func makeKeyRewriter(
 	sort.Slice(prefixes.rewrites, func(i, j int) bool {
 		return bytes.Compare(prefixes.rewrites[i].OldPrefix, prefixes.rewrites[j].OldPrefix) < 0
 	})
+	fromSystemTenant := false
+	for i := range tenants {
+		if tenants[i].OldID.IsSystem() && tenants[i].NewID.IsSystem() {
```

Comment:
can we add a comment just tying this back to the comment in the `restore_job` where we add this noop rekey to signal that the backup is from a system tenant? Another thought, what if we define these sentinel rekeys as consts to be even more explicit.

const (
    poisonPillRekey := ...
    isBackupFromSystemTenantRekey :=
)

and then compare `tenants[i] == isBackupFromSystemTenantRekey`? Feel free to ignore.

Reply:
there's a `[]byte` in tablerekey which makes it not comparable so I'll stick to checking the ID for that one, but I moved
the identity rewrite indicator for system into a named var used on both sides.